### PR TITLE
Item collision

### PR DESCRIPTION
**Pull request by jmithani**
Assigning @andrea-3000 
### What's this PR do?
- Adds collision and overlap detection for items.
- When the mouse hits an item, the game pauses, a dialogue bubble comes up, then the game resumes and the item disappears.
### Where should the reviewer start?
- Look in the `update()` function to see where the collisions are being checked, and then see the callback functions.

cc @jonuy looks like there are some conflicts

### CODE REVIEW

**Comment by jonuy** [comment]
Apologies in advance. I'm about to get more particular about code styles. Anywhere you see a :space_invader: is a comment about removing the extra line.

And this is probably for a later PR, but we should do spaces instead of tabs.

File: src/States/atDanceGameState.js [hunk 1]
```diff
@@ -10,6 +10,15 @@ function AtDanceGame() {
...
 
+	//items
+	var pickup;
+	var item1;
+	var item2;
+	var item3;
+	var itemDialogue1;
+	var itemDialogue2;
+	var itemDialogue3;
+
```
Comment (jonuy) [inline src/States/atDanceGameState.js]: :space_invader:

File: src/States/atDanceGameState.js [hunk 2]
```diff
@@ -28,11 +37,25 @@ function AtDanceGame() {
...
 
+		game.load.image('itemImg1', 'assets/ADG_player.png');
+		game.load.image('itemImg2', 'assets/ADG_player.png');
+		game.load.image('itemImg3', 'assets/ADG_player.png');
+
+		//item dialogue
+		game.load.image('itemDialogue1', 'assets/ADG_couple.png');
+		game.load.image('itemDialogue2', 'assets/ADG_couple.png');
+		game.load.image('itemDialogue3', 'assets/ADG_couple.png');
+
```
Comment (jonuy) [inline src/States/atDanceGameState.js]: :space_invader: This line and the next

File: src/States/atDanceGameState.js [hunk 3]
```diff
@@ -28,11 +37,25 @@ function AtDanceGame() {
...
+
 	}
 
 	console.log("loaded assets");
 
 	function create() {
+
```
Comment (jonuy) [inline src/States/atDanceGameState.js]: :space_invader:

File: src/States/atDanceGameState.js [hunk 4]
```diff
@@ -28,11 +37,25 @@ function AtDanceGame() {
...
+		game.physics.startSystem(Phaser.Physics.ARCADE);
+
```
Comment (jonuy) [inline src/States/atDanceGameState.js]: :space_invader: Basically just one new line between `game.physics.startSystem(Phaser.Physics.ARCADE);` and `//draws tables`

File: src/States/atDanceGameState.js [hunk 5]
```diff
@@ -41,12 +64,29 @@ function AtDanceGame() {
 
 		//draws table
 		player = game.add.sprite(createPlayer.x, createPlayer.y, 'playerImg');
+		game.physics.arcade.enable(player);
+
+
```
Comment (jonuy) [inline src/States/atDanceGameState.js:78]: :space_invader: Only need one new line here

File: src/States/atDanceGameState.js [hunk 6]
```diff
@@ -41,12 +64,29 @@ function AtDanceGame() {
...
+		item1 = game.add.sprite(100, 100, 'itemImg1');
+		game.physics.arcade.enable(item1);
+		item2 = game.add.sprite(200, 200, 'itemImg2');
+		game.physics.arcade.enable(item2);
+		item3 = game.add.sprite(300, 300, 'itemImg3');
+		game.physics.arcade.enable(item3);
 
 	}
 
 	function update() {
```
Comment (jonuy) [inline src/States/atDanceGameState.js:97]: :space_invader: No need for the new line spacing at the start and end of `update()`

File: src/States/atDanceGameState.js [hunk 7]
```diff
@@ -82,10 +122,55 @@ function AtDanceGame() {
 
 	}
 
+	function touchItem1() {
```
Comment (jonuy) [inline src/States/atDanceGameState.js]: :space_invader: New line spacing at the start and end of `touchItem1()`, `touchItem2()`, and `touchItem3()` aren't needed

**Comment by annie-digital** [comment]
:+1: :D

**Pull request merged**